client/extensions/woocommerce/app/promotions/promotion-form-card.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+import warn from 'lib/warn';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import promotionModels from './promotion-models';
+import SectionHeader from 'components/section-header';
+
+const promotionFieldEdit = ( siteId, promotion, editPromotion ) => ( fieldName, newValue ) => {
+	const newPromotion = {
+		type: promotion.type, // type always gets set because it's not set when selected.
+		[ fieldName ]: newValue,
+	};
+	editPromotion( siteId, promotion, newPromotion );
+};
+
+function renderFields( promotion, edit, translate, currency ) {
+	const model = promotionModels[ promotion.type ];
+
+	if ( ! model ) {
+		warn( 'No model found for promotion type: ' + promotion.type );
+		return null;
+	}
+
+	return Object.keys( model ).map( ( fieldName ) => {
+		const fieldModel = model[ fieldName ];
+		return renderField( fieldName, fieldModel, promotion, edit, currency );
+	} );
+}
+
+function renderField( fieldName, fieldModel, promotion, edit, currency ) {
+	const { component, labelText, explanationText, placeholderText, isRequired } = fieldModel;
+
+	return React.createElement( component, {
+		key: fieldName,
+		fieldName,
+		labelText,
+		explanationText,
+		placeholderText,
+		isRequired,
+		value: promotion[ fieldName ],
+		edit,
+		currency,
+	} );
+}
+
+function getHeaderText( promotionType, translate ) {
+	switch ( promotionType ) {
+		case 'product_sale':
+			return translate( 'Product & Sale Price' );
+		case 'fixed_product':
+		case 'fixed_cart':
+		case 'percent':
+			return translate( 'Coupon code & Discount' );
```

Comment:
Should code be capitalized like all the words in Product & Sale Price are?

Reply:
👍